packages/client/components/TransactionList.tsx

```diff
@@ -3,23 +3,25 @@ import { Text, Box } from "native-base";
 import { StyleSheet } from "react-native";
 import { Transaction } from "@cryptify/common/src/domain/entities/transaction";
 import { TransactionListItem } from "./TransactionListItem";
+import { CompositeNavigationProp } from "@react-navigation/native";
 
 type Props = {
     transactions: Transaction[];
     walletAddress: string;
     displaySeparation: boolean;
+    navigation: CompositeNavigationProp<any, any>;
 };
 
-export function TransactionList({ transactions, walletAddress, displaySeparation }: Props) {
+export function TransactionList({ transactions, walletAddress, displaySeparation, navigation }: Props) {
     let savedDate = new Date();
-    function renderHeader(date: Date) {
+    function renderHeader(date: Date, i: number) {
         if (savedDate?.getFullYear() == date.getFullYear() && savedDate.getMonth() == date.getMonth()) {
             savedDate = date;
             return;
         }
         savedDate = date;
         return (
-            <Box backgroundColor="text.100">
+            <Box key={i} backgroundColor="text.100">
```

Comment:
Since the header is for just unique months, just convert the first date of that month to a unix timestamp and use that, assuming im understanding correctly how this component works, if that works you can clean up all that mess below

```suggestion
            <Box key={+date} backgroundColor="text.100">
```